src/compile/Model.ts

```diff
@@ -233,48 +233,88 @@ export class Model {
     return 30;
   }
 
-  /** @return an object with format and formatType properties. */
+  /**
+   * Builds an object with format and formatType properties.
+   *
+   * @param format explicitly specified format
+   */
   public formatMixins(channel: Channel, format: string) {
     const fieldDef = this.fieldDef(channel);
 
     let def: any = {};
 
-    if (fieldDef.type === TEMPORAL) {
-      // explicitly set the fromat type so that vega uses the datetime formatter
-      def.formatType = 'time';
+    switch (fieldDef.type) {
+      case QUANTITATIVE:
+        def.formatType = 'number';
+        break;
+      case TEMPORAL:
+        def.formatType = 'time';
+        break;
     }
 
     if (format !== undefined) {
       def.format = format;
-      return def;
+    } else {
+      switch (fieldDef.type) {
+        case QUANTITATIVE:
+          def.format = this.numberFormat(channel);
+          break;
+        case TEMPORAL:
+          const f = this.timeFormat(channel);
```

Comment:
`def.format = this.timeFormat(channel) || this.config().timeFormat`
is more concise?